packages/grafana-ui/src/components/DataSourceSettings/AlertingSettings.tsx

```diff
@@ -1,13 +1,34 @@
-import { DataSourcePluginOptionsEditorProps } from '@grafana/data';
-import React from 'react';
+import { DataSourceInstanceSettings, DataSourceJsonData, DataSourcePluginOptionsEditorProps } from '@grafana/data';
+import React, { useMemo } from 'react';
 import { Switch } from '../Forms/Legacy/Switch/Switch';
+import { InlineField } from '../Forms/InlineField';
+import { InlineFieldRow } from '../Forms/InlineFieldRow';
+import { Select } from '../Select/Select';
 
-type Props<T> = Pick<DataSourcePluginOptionsEditorProps<T>, 'options' | 'onOptionsChange'>;
+interface Props<T> extends Pick<DataSourcePluginOptionsEditorProps<T>, 'options' | 'onOptionsChange'> {
+  alertmanagerDataSources: Array<DataSourceInstanceSettings<DataSourceJsonData>>;
+}
+
+interface AlertingConfig extends DataSourceJsonData {
+  manageAlerts?: boolean;
+}
 
-export function AlertingSettings<T extends { manageAlerts?: boolean }>({
+export function AlertingSettings<T extends AlertingConfig>({
+  alertmanagerDataSources,
   options,
   onOptionsChange,
 }: Props<T>): JSX.Element {
+  const alertmanagerOptions = useMemo(
+    () =>
+      alertmanagerDataSources.map((ds) => ({
+        label: ds.name.substr(0, 37),
+        value: ds.name,
```

Comment:
Consider using datasource id instead of name here, so that renaming the alertmanager datasource wouldn't break the link. 
Also we might need to consider what happens if it's deleted... Maybe we can add some code to backend to remove links if alertamnager is deleted? 
Perhaps worth to investigate how Prom/Loki -> Tempo link is handled